src/query/api/v1/handler/database/create.go

```diff
@@ -188,43 +231,132 @@ func (h *createHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	xhttp.WriteProtoMsgJSONResponse(w, resp, logger)
 }
 
-func (h *createHandler) parseRequest(r *http.Request) (*admin.NamespaceAddRequest, *admin.PlacementInitRequest, *xhttp.ParseError) {
+func (h *createHandler) maybeInitPlacement(
+	currPlacement clusterplacement.Placement,
+	parsedReq *admin.DatabaseCreateRequest,
+	placementRequest *admin.PlacementInitRequest,
+	r *http.Request,
+) (clusterplacement.Placement, bool, error) {
+	if currPlacement == nil {
+		// If we're here then there is no existing placement, so just create it. This is safe because in
+		// the case where a placement did not already exist, the parse function above validated that we
+		// have all the required information to create a placement.
+		newPlacement, err := h.placementInitHandler.Init(placement.M3DBServiceName, r, placementRequest)
+		if err != nil {
+			return nil, false, err
+		}
+
+		return newPlacement, false, nil
+	}
+
+	// NB(rartoul): Pardon the branchiness, making sure every permutation is "reasoned" through for
+	// the scenario where the placement already exists.
+	switch dbType(parsedReq.Type) {
+	case dbTypeCluster:
+		if placementRequest != nil {
+			// If the caller has specified a desired clustered placement and a placement already exists,
+			// throw an error because the create database API should not be used for modifying clustered
+			// placements. Instead, they should use the placement APIs.
+			return nil, true, errClusteredPlacementAlreadyExists
+		}
+
+		if placementIsLocal(currPlacement) {
+			// If the caller has specified that they desire a clustered placement (without specifying hosts)
+			// and a local placement already exists then throw an error because we can't ignore their request
+			// and we also can't convert a local placement to a clustered one.
+			return nil, true, errCantReplaceLocalPlacementWithClustered
+		}
+
+		// This is fine because we'll just assume they want to keep the same clustered placement
```

Comment:
worth logging this so that the user knows?

Reply:
the placement comes back in the API response, so I think its ok